bin/main.rb

```diff
@@ -1,2 +1,82 @@
 #!/usr/bin/ruby
-puts 'Hello world!'
+
+@data = [[' ', ' ', ' '], [' ', ' ', ' '], [' ', ' ', ' ']]
+
+def user_info
+  puts "Please introduce first player's name:"
+  @player1 = gets.chomp
+  puts "Please introduce second player's name:"
+  @player2 = gets.chomp
+end
+
+def random_winner
+  random = rand
+  puts "Player #{@player1} has won!" if random < 0.2
+  puts "Player #{@player2} has won!" if random > 0.2 && random < 0.4
+  puts 'This is a Draw!' if random > 0.4
+end
+
+def iteration
+  user_info
+  board_to_s
+  i = 0
+  player = @player1
+  symbol = 'X'
+  while i < 9
```

Comment:
- [ ] [REQUIRED] The use of an out-of-nowhere number to terminate a loop here is a signal that something is wrong.
**Hint**: Use a terminating variable instead. An example is shown below where `game_on` is the terminating variable.
```
game_on = true

# initial instructions for player

while game_on
  # loop for each move

  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```

Reply:
Hi @krys2fa. Thank you for your review.
We fixed the linter error regarding the length of the block, and we removed the local variable in the loop.
But we can not fix the frozen literal linter error because if we add the comment to enable it, then we get a linter error on Github.

Reply:
@Zappat0n, what was the GitHub Action error message?

Reply:
![Screenshot from 2020-11-25 22-31-41](https://user-images.githubusercontent.com/64755700/100283783-63128c00-2f6e-11eb-9c03-2db651791ad9.png)